File: benchmark/comparison/results/blind_py/28.py
class Ok:
    def __init__(self, v):
        self.v = v
    def __repr__(self):
        return f"Ok({self.v})"

class Err:
    def __init__(self, msg):
        self.msg = msg
    def __repr__(self):
        return f"Err({self.msg})"

def safe_div(a, b):
    if b == 0:
        return Err("division by zero")
    return Ok(int(a / b))

def chain_div(a, b, c):
    r = safe_div(a, b)
    if isinstance(r, Err):
        return r
    return safe_div(r.v, c)

print(safe_div(10, 2))
print(safe_div(10, 0))
print(safe_div(7, 2))
print(chain_div(100, 5, 4))
print(chain_div(100, 0, 4))
print(chain_div(100, 5, 0))

File: benchmark/comparison/results/blind_ts/28.ts
type Result = { tag: "Ok"; value: number } | { tag: "Err"; msg: string };

function Ok(value: number): Result { return { tag: "Ok", value }; }
function Err(msg: string): Result { return { tag: "Err", msg }; }

function formatResult(r: Result): string {
  if (r.tag === "Ok") return `Ok(${r.value})`;
  return `Err(${r.msg})`;
}

function safe_div(a: number, b: number): Result {
  if (b === 0) return Err("division by zero");
  return Ok(Math.trunc(a / b));
}

function chain_div(a: number, b: number, c: number): Result {
  const r1 = safe_div(a, b);
  if (r1.tag === "Err") return r1;
  return safe_div(r1.value, c);
}

console.log(formatResult(safe_div(10, 2)));
console.log(formatResult(safe_div(10, 0)));
console.log(formatResult(safe_div(7, 2)));
console.log(formatResult(chain_div(100, 5, 4)));
console.log(formatResult(chain_div(100, 0, 4)));
console.log(formatResult(chain_div(100, 5, 0)));
